kratos/utilities/parallel_levelset_distance_calculator.h

```diff
@@ -148,153 +151,116 @@ class ParallelDistanceCalculator
             }
 
-
             //finalize the computation of the distance
-            #pragma omp parallel for private(DN_DX)
-            for(int i = 0; i<node_size; i++)
-            {
-                ModelPart::NodesContainerType::iterator it=rModelPart.NodesBegin()+i;
-                double& area = it->FastGetSolutionStepValue(rAreaVar);
-                double& is_visited = it->GetValue(IS_VISITED);
-                if(area > 1e-20 && is_visited != 1.0) //this implies that node was computed at the current level and not before
+            block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+                double& r_area = rNode.FastGetSolutionStepValue(rAreaVar);
+                if(r_area > 1e-20 && rNode.IsNot(VISITED)) //this implies that node was computed at the current level and not before
                 {
-                    double& distance = it->FastGetSolutionStepValue(rDistanceVar);
-                    distance /= area;
-                    is_visited = 1.0;
+                    double& r_distance = rNode.FastGetSolutionStepValue(rDistanceVar);
+                    r_distance /= r_area;
+                    rNode.Set(VISITED, true);
                 }
-            }
-
+            });
         }
 
-        //*****************************************************************+
-        //*****************************************************************+
-        //*****************************************************************+
         //assign the sign to the distance function according to the original distribution. Set to max for nodes that were not calculated
-        #pragma omp parallel for
-        for(int i = 0; i<node_size; i++)
-        {
-            ModelPart::NodesContainerType::iterator it=rModelPart.NodesBegin()+i;
-            const double area = it->FastGetSolutionStepValue(rAreaVar);
-            double& dist = it->FastGetSolutionStepValue(rDistanceVar);
-
-
-            if(dist > max_distance || area <1e-20)
-                dist = max_distance;
+        block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+            const double area = rNode.FastGetSolutionStepValue(rAreaVar);
+            double& r_dist = rNode.FastGetSolutionStepValue(rDistanceVar);
+            if(r_dist > MaxDistance || area < 1e-20) {
+                r_dist = MaxDistance;
+            }
 
-//            if(it->GetValue(IS_FLUID) == 1.0)
-//                dist = -fabs(dist);
-//            else
-//                dist = fabs(dist);
-        }
+            // if (rNode.Is(FLUID)) {
+            //     r_dist = -std::abs(r_dist);
+            // } else {
+            //     r_dist = std::abs(r_dist);
+            // }
+        });
 
         KRATOS_CATCH("")
     }
 
-
-
-
     //**********************************************************************************
     //**********************************************************************************
     double FindMaximumEdgeSize(ModelPart& r_model_part)
```

Comment:
I am guessing this method will be removed, probably here you could also use the parallel utils

Reply:
I think we will remove it in a future PR. If this is not the case, I'd do it.